src/bun.js/webcore/fetch.zig

```diff
@@ -1516,11 +1516,19 @@ pub fn Bun__fetch_(
     bun.analytics.Features.fetch += 1;
     const vm = jsc.VirtualMachine.get();
 
-    var memory_reporter = bun.handleOom(bun.default_allocator.create(bun.MemoryReportingAllocator));
+    var memory_reporter: *bun.MemoryReportingAllocator = undefined;
```

Comment:
let's make it an optional pointer default initialized to null 
```suggestion
    var memory_reporter: ?*bun.MemoryReportingAllocator = null;
```